scheds/rust/scx_layered/src/bpf/util.bpf.c

```diff
@@ -54,14 +68,33 @@ static char *format_cgrp_path(struct cgroup *cgrp)
 	return path;
 }
 
-static inline bool match_prefix(const char *prefix, const char *str, u32 max_len)
+bool __noinline match_prefix(const char *prefix, const char *str, u32 max_len)
 {
-	int c;
+	u32 c, zero = 0;
+	int len;
+
+	if (!prefix || !str || max_len > MAX_PATH)
+		return false;
```

Comment:
Would it be better to trigger `scx_bpf_error()` instead of failing silently? This and the following sanity checks shouldn't trigger and it probably is better to be explicit when such failures happen.